src/middlewares/checkIfExists.js

```diff
@@ -0,0 +1,23 @@
+const expenseService = require('../services/expense.service');
+const userService = require('../services/user.service');
+
+const checkIfExists = (entryType) => {
+  return async (req, res, next) => {
+    const { id } = req.params;
+    const entryServices = {
+      user: userService,
+      expense: expenseService,
+    };
+
+    const entry = await entryServices[entryType].getById(id);
```

Comment:
Consider adding error handling for the `getById` call to manage potential exceptions, such as database connection errors, which could prevent the middleware from functioning correctly.